config/routes.rb

```diff
@@ -5,8 +5,12 @@
     resources :suppliers
     resources :reports, only: [:index] do
       collection do
-        get   :earnings
-        post  :earnings
+        get :earnings
+        post :earnings
```

Comment:
Interesting, is the current report sending the form as a post request? this should be `GET` so it can be bookmarkable